taxi/views.py

```diff
@@ -86,4 +87,30 @@ class DriverListView(LoginRequiredMixin, generic.ListView):
 
 class DriverDetailView(LoginRequiredMixin, generic.DetailView):
     model = Driver
-    queryset = Driver.objects.all().prefetch_related("cars__manufacturer")
+    queryset = Driver.objects.prefetch_related("cars__manufacturer")
+
+
+class DriverCreateView(LoginRequiredMixin, generic.CreateView):
+    model = Driver
+    form_class = DriverCreationForm
+
+
+class DriverLicenseUpdateView(LoginRequiredMixin, generic.UpdateView):
+    model = Driver
+    form_class = DriverLicenseUpdateForm
+    success_url = reverse_lazy("taxi:driver-list")
+
+
+class DriverLicenseDeleteView(LoginRequiredMixin, generic.DeleteView):
+    model = Driver
+    success_url = reverse_lazy("taxi:driver-list")
+
+
+def assign_unassign_to_car(request, *args, **kwargs):
+    car_pk = kwargs["pk"]
+    user = request.user.cars
+    if user.filter(id=car_pk):
+        user.remove(car_pk)
+    else:
+        user.add(kwargs["pk"])
```

Comment:
u've extracted this `kwargs["pk"]` to `car_pk` variable, so why don't you use it here ?

Reply:
Fixed